### ISSUE TITLE
Register SpineInterface as a Julia package

### ISSUE THREAD

**Issue by clizbe** [spine-tools/SpineInterface.jl#117]
This is something we've been discussing on the SpineOpt side. If we can get Spine into the Julia registry, we can use the Julia Registrator bot to register more frequently and hopefully avoid a lot of bugs with installs and updates.

For us to register SpineOpt, first SpineInterface has to be registered (since it's a dependency).

Here are the guidelines for registering a new package:
https://juliaregistries.github.io/RegistryCI.jl/stable/guidelines/

Once all those boxes are checked, here's the procedure:
![image](https://github.com/spine-tools/SpineInterface.jl/assets/11889283/d60d5840-29f9-4330-a43a-17b95a36ea6d)

Then Julia will review the repository and give feedback or approve it.

If you have trouble, please tag @abelsiqueira

**Comment by clizbe** [comment]
@manuelma @jkiviluo Do we agree we should do this? If so I'll try to do it and then will do SpineOpt.

**Comment by clizbe** [comment]
@jkiviluo @manuelma 
This repo needs an upper bound on the julia version compat (see #4 below). The repo says "Julia 1.8 or later." In the Project.toml it says "^1.8" which I guess means 1.8 or later? I'm afraid to set it to the most recent julia release in case this causes problems for people.
![image](https://github.com/user-attachments/assets/a10475c3-df9c-4e16-b74e-9282abfc1f33)

Here's the guidance from Julia:
![image](https://github.com/user-attachments/assets/6dfe9950-9116-4ecc-90e3-c3211fa7b340)

**Comment by abelsiqueira** [comment]
The "^1.8" means 1.8 up to, and excluding, 2.
Ref: https://pkgdocs.julialang.org/v1/compatibility/#Caret-specifiers

**Comment by abelsiqueira** [comment]
Yes! General is for us as well. You can follow the process in JuliaHub (I never used it, but I heard it's good) or follow the steps we do with Tulipa.

I checked the package briefly, and I noticed other stuff missing. They are not required for the registration, but they will simplify the next steps. If you don't want to delay the registration, these can be added later. If they're added before the registration is merged (3 days), then it probably won't be an issue, and they're only Quality of Life improvements, not required.

- Add a TagBot.yml - After merging, it will automatically create the tag/release on GitHub.
- Confirm that it has (or add) a DOCUMENTER_KEY (https://abelsiqueira.com/BestieTemplate.jl/stable/10-full-guide/#Documentation-for-releases) - After a tag/release is created on GitHub, it will trigger the Documentation build. The DOCUMENTER_KEY is required to push the docs online.

**Comment by clizbe** [comment]
Tagbot.yml - Can I just copy the one from Tulipa?
Documenter_key - In those instructions when it says Username should I put spine-tools? Because I put my Github account but now the directions include my fork of the repo rather than the actual repo.

**Comment by abelsiqueira** [comment]
Yes and yes

**Comment by clizbe** [comment]
Oh Manuel already added the Documenter key for this repo. :)

**Comment by clizbe** [comment]
"The following dependencies do not have a `[compat]` entry that is upper-bounded and only includes a finite number of breaking releases: DataStructures, JSON, PyCall, Requires, URIs\n

**Comment by abelsiqueira** [comment]
The section `[compat]` does not have a corresponding value that appears on `[deps]`.
There is no easy way to know the actual minimally required version, so what you can do is check the currently installed versions.
Go to `SpineInterface.jl`, run `Julia`, activate the package, enter `pkg>` mode and run `status`.
The packages that don't appear in the status are from the standard library and can be ignored for now.

Also, post a link to the registration PR, please.

**Comment by clizbe** [comment]
Here's the PR:
https://github.com/JuliaRegistries/General/pull/116052#issuecomment-2376971783

**Comment by clizbe** [comment]
> The section `[compat]` does not have a corresponding value that appears on `[deps]`. There is no easy way to know the actual minimally required version, so what you can do is check the currently installed versions. Go to `SpineInterface.jl`, run `Julia`, activate the package, enter `pkg>` mode and run `status`. The packages that don't appear in the status are from the standard library and can be ignored for now.
> 
> Also, post a link to the registration PR, please.

Yeah I was trying this to see what I should add to the [compat] section but it's kind of blank:
![image](https://github.com/user-attachments/assets/e1c020ef-f808-4bb6-b9af-05d373d5b220)

**Comment by clizbe** [comment]
I ran pkg>up and got this, but I don't know if these are legit to add to the Project.toml without breaking anything:
![image](https://github.com/user-attachments/assets/4a8e038d-3bb9-413d-8f3c-76d665fd613c)

**Comment by clizbe** [comment]
Btw is this the thing compat_helper is always updating for us? 🤔

**Comment by abelsiqueira** [comment]
You can't find the minimum compat easily. You would need to  try decreasing versions and running things until they break.
That being said it shouldn't break things **online** (assuming test pass), but if people have older versions locally they'll have to update everything. You can also ask other people for their versions so it's safer, but I don't think it's a big deal.
You can also install compathelper, run it, and merge the PRs. It is indeed what's always updating for us (yay automation).

**Comment by clizbe** [comment]
Yeah I think adding compathelper is a good idea. (Really we should just do BestieTemplate.) And I don't think it's a huge problem if people have to update, right? They're updating all the time anyway I think...

**Comment by abelsiqueira** [comment]
Without knowing how to build SpineInterface/Opt/db, I can't be sure, but my guess is that it shouldn't be a problem. But if updating is a problem, I would take the change to update the documentation of the build process (and/or the build process itself) to make it less painful.
+1 for Bestie, of course. You could apply the "Minimum" Bestie, which includes the TagBot. But it also includes a few changes that are not trivial - such as changing the test and docs workflows.

**Comment by clizbe** [comment]
Yeah I think applying Bestie would be nice in the future but I'd need quite some help with it.

I'm planning on updating the docs, both for install and releases - but first need to check the JuliaRegistry version works...

**Comment by clizbe** [comment]
We've decided we're going to:
- [ ] Use the latest version as the lower bound and see if people complain their stuff breaks
   - [ ] Maybe make an announcement that people need to update their dependencies
- [ ] Add compathelper
- [ ] Finish registering both Interface & Opt
- [ ] Check that new registery versions work
- [ ] Update docs:
  - [ ] Dev: How to release new version
  - [ ] Install: How to remove SpineRegistry from computer
  - [ ] Install: How to install from JuliaRegistry 
- [ ] Announcement! Important because SpineRegistry version will no longer exist/receive updates

**Comment by clizbe** [comment]
Here's the new PR:
https://github.com/JuliaRegistries/General/pull/116905
